### DISCUSSION TITLE
Leakage-Safe Multimodal Biosignal Fusion for Choice Decoding and AIxBio Inference-Risk

### DISCUSSION THREAD

**Discussion by aaygan29** [AI4S-Bench/ai4s-benchmark#22] [Task Proposals]
## Scientific Domain

Biology, AI / ML, Applied Mathematics, Interdisciplinary > Computational neuroscience; neuroforecasting; biosignal decoding; neuroprivacy / AIxBio safety.

## Scientific Problem
Behaviour and decisions leave traces in the body: cardiac, electrodermal, pupillary, respiratory, endocrine, and neural signals. Each trace is a weak predictor on its own. The scientific and safety question is what happens when an AI system is given several of them at once under one identity.

$$s_m = \ell + \varepsilon_m$$
where $s_m$ is the read from family $m$, $\ell$ is the shared latent decision signal, and $\varepsilon_m$ is family-specific noise.

Formally, each biosignal family gives a noisy read of the same latent decision signal:

$$\sigma_{\text{fused}}^2 = \frac{1}{\sum_{m} 1/\sigma_m^2} \le \min_{m} \sigma_m^2$$

If errors are independent, the inverse-variance combination has variance, so the fused estimate beats the best single family whenever more than one family carries real, independent signal. This inequality is proved in Lean 4 / Mathlib in the prior work and checked empirically.

The task asks the agent to:

Build or use a pipeline with one model per biosignal family, then reconcile them by majority, soft mean, and out-of-fold accuracy weighting.

Evaluate with subject-grouped nested CV, in-fold SMOTE inside bagging, balanced accuracy, ROC AUC, Brier/ECE, subject-level bootstrap CIs, and label-permutation tests.

Reproduce the synthetic positive control: six planted families, fusion beats the best single family, the ablation ladder rises until a pure-noise family is appended, the noise family gets weight 0, and the deliberately weak endocrine family fails its permutation test as planted.

Reproduce real single-family leaks on public data: NARPS gambles, ds002989 delay-discounting bids, ASZED-153 EEG, Kiva microloans, and PPG-BP.

Project the value of a fourth sensor with the calibrated fusion-gain script, reporting a band over error correlation 
ρ
ρ, not a single flattering number.

Test the "chained latent" hypothesis and report the null honestly across seeds.

Produce a reproducible run record: resolved config, git commit and dirty flag, per-family provenance, both evaluation levels, bootstrap intervals, permutation p-values, calibration curves, and final weights.

Why it matters scientifically: this turns a set of individually harmless-looking leaks into a strong inference over protected behavioural and health attributes. An AI with standing access to fused biosignal streams is effectively an inference engine over the body. The defensive claim is not "do not sense biosignals"; it is that aggregate access to multiple biosignal streams under one identity should be governed as a capability. The concern is sharpest for populations whose data is collected with the least protection, including the Global South.

The classic reward-anticipation neuroforecasting result is used as one input among several biosignal drivers, not as the thesis. The thesis is the AIxBio safety one.

## Solvability
Solvable in principle. The synthetic generator provides a positive control with known planted structure. Public real datasets and event files are available. The difficulty comes from the science and statistics, not from missing data:

subject leakage must be prevented;

resampling leakage must be prevented;

EEG covariance must be handled on the SPD manifold, e.g. Riemannian tangent-space classification;

small real n requires subject-level bootstrap intervals;

within-subject label permutation is degenerate for diagnosis and loan-level labels, and must be explained rather than hidden;

fusion weights must come from out-of-fold skill, not training accuracy;

correlated arousal drivers mean fusion gain must be projected over ρ, not assumed independent.

A capable agent can solve this by using the provided package or reimplementing the pipeline, then passing the runtime gates and producing the run records.

## References & Resources
Core papers

Genevsky, Yoon & Knutson 2017 — neuroforecasting. DOI: 10.1523/JNEUROSCI.1633-16.2017

Saeb et al. 2017 — record-wise vs subject-wise CV on wearable data. DOI: 10.1093/gigascience/gix019

Chawla et al. 2002 — SMOTE. DOI: 10.1613/jair.953

Sci Rep 2024 — resample-inside-CV rule. DOI: 10.1038/s41598-024-62585-z

Breiman 1996 — bagging. DOI: 10.1007/BF00058655

Barachant et al. 2012 — Riemannian tangent-space EEG. DOI: 10.1109/TBME.2011.2172210

Yger, Berar & Lotte 2017 — Riemannian review. DOI: 10.1109/TNSRE.2016.2627016

Arsigny et al. 2006 — log-Euclidean metric. DOI: 10.1002/mrm.20965

Brodersen et al. 2010 — balanced accuracy. DOI: 10.1109/ICPR.2010.764

Ojala & Garriga 2010 — permutation tests. ACM

Forte et al. 2021 — cardiac/HRV. DOI: 10.3390/brainsci11020243

Dunn et al. 2006 — electrodermal somatic marking. DOI: 10.1016/j.neubiorev.2005.07.001

Pajkossy et al. 2017 — pupil/locus-coeruleus. DOI: 10.1111/psyp.12964

Vincent et al. 2019 — pupil/LC computational. DOI: 10.1371/journal.pcbi.1007126

Coates & Herbert 2008 — endocrine risk bias. DOI: 10.1073/pnas.0704025105

Cueva et al. 2015 — endocrine risk bias. DOI: 10.1038/srep11206

Zelano et al. 2016 — respiration. DOI: 10.1523/JNEUROSCI.2586-16.2016

Datasets

OpenNeuro ds001734 — NARPS gambles.

OpenNeuro ds002989 — delay-discounting bids.

ASZED-153 EEG — Nigeria.

Kiva microloans — Kenya.

PPG-BP fingertip pulse.

### Software and tools
Software and tools
Language and core stack.

Python 3.11+ (required).

NumPy, SciPy, scikit-learn, pandas, matplotlib, pytest, ruff.

Install via pip install -e ".[dev]" from the repository root.

Optional biosignal backends (install only when the corresponding family is exercised).

pip install -e ".[fmri]" — nilearn, nibabel: fMRI loading and ROI extraction.

pip install -e ".[eeg]" — MNE: EEG loading and preprocessing.

pip install -e ".[riemann]" — pyriemann: affine-invariant Riemannian tangent-space backend for EEG covariance classification.

pip install -e ".[face]" — opencv-python: action-unit and landmark features from facial video.

pip install -e ".[vision]" — torch, transformers: the real ViT face encoder.

Formal verification (optional gate).

Lean 4 with Mathlib, to build proofs/FusionMath.lean.

lake build must exit 0 with no sorry.

CLI entry points the agent must be able to invoke.

bash
python scripts/run_demo.py --quick
python scripts/run_biosignal_fusion.py --quick
python scripts/project_fusion_gain.py --validate
python scripts/run_narps_behavior.py --root data/raw/narps_events
python scripts/run_delay_discounting.py --root data/raw/ds002989

Test and lint commands.

bash
pytest
pytest tests/test_fusion_math.py -q
pytest tests/test_leakage.py -v
ruff check src tests scripts

No GPU is required for the core pipeline. CPU-only execution is the intended path. A GPU becomes useful only if the agent enables the optional ViT face encoder under .[vision], which is not part of the scored gates.

Internet access. Not available during evaluation. All datasets, cached features, and model weights must be pre-staged locally.

### Dataset & artifacts
Provided to the agent (open, pre-cached, licensed).

Artifact	Provenance	License / terms	Role in the task
Synthetic generator (src/behavioral_decoding/synthetic.py)	Authored in-repo	Repository license	Positive control with six planted biosignal families and known independent structure
NARPS events.tsv (ds001734)	OpenNeuro	OpenNeuro dataset license	27,454 gambles; accept/reject behaviour
Delay-discounting events.tsv (ds002989)	OpenNeuro	OpenNeuro dataset license	4,134 bids; lowball-this-offer behaviour
ASZED-153 EEG (Nigeria)	Published cohort	Original cohort terms	76 subjects; undisclosed diagnosis target
Kiva microloans (Kenya)	Public loan records	Original terms	3,023 loans; repayment target
PPG-BP fingertip pulse	Public dataset	Original terms	219 subjects; hypertension, diabetes, sex targets
Must stay hidden from the agent.

Held-out synthetic seeds used for scoring.

Hidden test splits and the frozen evaluation harness.

Reference run records, reference figures, and reference JSON outputs.

The scoring rubric implementation and any golden metric files.

Any future cohort data (e.g. DEAP, WESAD) reserved for a later stage.

What the agent must not do.

Modify hidden test data or the evaluation harness.

Claim imaging results from the OpenNeuro arms. Those archives contain real events.tsv files but git-annex pointer stubs where BOLD volumes should be, so no imaging is used and no imaging claim is permitted.

Report a result whose configuration is not recorded in a run record.

What the agent produces as artifacts.

Code patch or notebook implementing or invoking the pipeline.

JSON run records containing: resolved config, git commit and dirty flag, per-family provenance, both evaluation levels, bootstrap intervals, permutation 
p
p-values, calibration curves, and final weights.

Figures: per-family accuracy, fusion ladder, projection calibration.

A final report stating honest limits.

Total real human subjects across provided arms. 443 people, including African clinical EEG and Kenyan borrower records.

### Computation resources (time and device)
**Budget envelope (one GPU, 16 CPU cores max).**

| Resource | Requirement |
|---|---|
| CPU | 4–16 cores |
| RAM | 4–16 GB |
| GPU | Not required; optional ViT only |
| Disk | ~20 GB |
| Quick mode | ~10–20 min |
| Full mode | ~2–4 hours |
| Lean build | ~10–30 min |

**Notes.**

- Quick mode: pipeline + gates on synthetic data.
- Full mode: reproduce published numbers and Africa-arm tables.
- No network; pre-stage datasets and weights.
- Scored gates are CPU-only.

### Expected workflow & outputs
Starting point.

A clean checkout of aaygan29/behavioral_decoding with the base environment installed via pip install -e ".[dev]".

Pre-cached datasets at data/raw/narps_events and data/raw/ds002989, plus processed ASZED-153, Kiva, and PPG-BP features.

No run records, figures, or results present. The agent generates them.

Expected workflow.

Install the base environment and confirm pytest and ruff check pass on the untouched repository.

Run scripts/run_demo.py --quick to confirm the end-to-end pipeline executes on synthetic data.

Run scripts/run_biosignal_fusion.py --quick. Confirm the per-family baselines, the fused result, the ablation ladder, and the five runtime gates. The script exits non-zero if any gate fails.

Run the real behavioural arms: scripts/run_narps_behavior.py and scripts/run_delay_discounting.py.

Run the processed real arms for ASZED-153, Kiva, and PPG-BP through the same pipeline.

Run scripts/project_fusion_gain.py --validate and confirm the projection calibration: predicted 
≈
0.645
≈0.645 under independence, closest at 
ρ
=
0.2
ρ=0.2, against measured 
0.623
0.623.

Run scripts/run_chained_fusion.py across three generator seeds and report the null contrast honestly.

Optionally build proofs/FusionMath.lean with lake build and confirm exit 0 with no sorry.

Emit JSON run records for every reported result, plus the three figures and a final report.

Hand-back (expected outputs).

A code patch or notebook.

One JSON run record per reported result, each containing the resolved config, git commit and dirty flag, per-family provenance, both evaluation levels, bootstrap intervals, permutation 
p
p-values, calibration curves, and final weights.

Three figures: per-family accuracy, fusion ladder, projection calibration.

A final report with an explicit honest-limits section, including the synthetic-positive-control caveat and the degenerate-permutation-test explanation for ASZED-153 and Kiva.

No imaging claim from the OpenNeuro event-only arms.

Open questions (left to the agent where no workflow is prescribed).

Choice of learner within the bagged per-family estimator, as long as the leakage guards in tests/test_leakage.py still pass.

Whether to use the pyriemann backend or the fallback tangent-map implementation for EEG covariance.

Whether to attempt the optional ViT face encoder, which is not part of the scored gates.

How to present the correlated-noise negative control alongside the main fusion result.

### How will this task be evaluated?
Metric type. Deterministic, script-based. No LLM judge is used for scoring. Every gate is a reproducible check against a fixed harness, fixed seeds, and schema-validated run records.

Primary metrics (computed by the pipeline, not by a judge).

Balanced accuracy and ROC AUC per family and for each fusion arm.

Brier score and ECE for calibration.

Subject-level bootstrap 95% CI, resampled by subject, not by trial.

Label-permutation 
p
p-value, with the within-subject shuffle explained where it is degenerate.

Fusion-minus-best-single delta.

Ablation ladder monotonicity.

Noise-family weight.

Projection calibration error against the measured value.

Required gates (pass/fail, all must hold).

pytest tests/test_leakage.py passes.

pytest tests/test_fusion_math.py passes.

Synthetic fusion beats the best single family for both soft vote and accuracy-weighted arms.

Ablation ladder rises as independent families join; flat when the noise family is appended.

Noise family weight 
≈
0
≈0.

Weak endocrine family fails its permutation test as planted, 
p
>
0.05
p>0.05.

Real single-family arms reproduce published balanced accuracy within tolerance: ds002989 
≈
0.882
≈0.882, NARPS 
≈
0.814
≈0.814, ASZED-153 
≈
0.705
≈0.705, Kiva 
≈
0.631
≈0.631, PPG-BP near chance.

Degenerate permutation tests correctly identified as tool/data mismatch; subject-level bootstrap CIs reported as the valid statistic.

Projection predicts 
≈
0.645
≈0.645 under independence, closest at 
ρ
=
0.2
ρ=0.2, against measured 
0.623
0.623.

Chained-fusion contrast is null across three seeds; sign flips; no reliable improvement.

Every reported result has a run record with resolved config, git commit, dirty flag, per-family provenance, both evaluation levels, bootstrap intervals, permutation 
p
p-values, calibration curves, and final weights.

No imaging claim from the OpenNeuro event-only arms.

Reproducibility of the measurement.

Random seeds fixed and recorded.

Subject-grouped splits, frozen across arms.

In-fold SMOTE only; no resampling before the split.

Bootstrap resamples subjects, not trials.

Runtime gates inside run_biosignal_fusion.py exit non-zero on failure.

Every number must be traceable to a run record. A result without a recorded config is treated as unreproduced and scores zero on that item.

Optional: lake build on proofs/FusionMath.lean exits 0 with no sorry.

Scoring rubric.

40
%
40% correctness and gate passage.

20
%
20% leakage-free methodology.

15
%
15% statistical calibration and uncertainty.

15
%
15% reproducibility and run-record completeness.

10
%
10% clarity, ethics, and honest limits.

Hidden evaluation.

Held-out synthetic seeds.

Real-data reruns on fixed splits.

Schema validation of run records.

Adversarial checks for subject leakage and resampling leakage.

### Is there risk of cheating and leakage?
What are the risks of cheating and leakage?

Public answers? Partly. The public repo contains the pipeline and published numbers, so an agent could copy outputs instead of computing them.

Mitigations. Held-out seeds, hidden splits, schema-validated run records with git commit, adversarial probes for subject overlap and pre-split SMOTE, and a required sign flip across seeds.

Weak resources. The public repo, OpenNeuro, ASZED-153, Kiva, PPG-BP, and the DOIs. Unavoidable for a reproduction, so defences are structural.

Leakage. Subject leakage, resampling leakage, weight leakage, and permutation degeneracy, each blocked in code or handled by subject-level bootstrap CIs.

Repeatable? Yes. Fixed seeds, fixed splits, frozen harness.

## Contributor

Name: Aayush Gandhi
Institution / affiliation: None provided
GitHub: https://github.com/aaygan29

## Task Metadata
Proposed task slug: `leakage-safe-multimodal-biosignal-fusion-for-choice-decoding-and-aixbio-inferenc`

---
Submitted via ai4sbench contribution form
